pzsd_bot/cogs/advent_of_code/leaderboard.py

```diff
@@ -0,0 +1,121 @@
+import logging
+
+import pendulum
+from discord import ApplicationContext, Bot, Embed
+from discord.commands import SlashCommandGroup, option
+from discord.ext.commands import Cog
+
+from pzsd_bot.settings import AOCSettings, Colors
+
+logger = logging.getLogger(__name__)
+
+CURRENT_YEAR = pendulum.today().year
+AOC_GENESIS = 2015
+
+
+class AOCLeaderboards(Cog):
+    aoc = SlashCommandGroup("aoc", "Advent of Code related commands.")
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.cached_leaderboards: dict = {}
+
+    def make_aoc_lb_embed(self, leaderboard: list) -> Embed:
+        ESC = "\x1b"
+        RESET = f"{ESC}[0m"
+        GOLD_BOLD = f"{ESC}[1;33m"
+        RED_BOLD = f"{ESC}[1;31m"
+        GREEN_BOLD = f"{ESC}[1;32m"
+        RED = f"{ESC}[31m"
+        GREEN = f"{ESC}[32m"
+
+        header = f"{GREEN_BOLD}Rank |  Stars  | Score | Name {RESET}"
+        divider = f"{RED_BOLD}-----+---------+-------+-------- {RESET}"
+
+        lines = [header, divider]
+
+        for rank, (score, stars, name) in enumerate(
+            sorted(leaderboard, reverse=True), 1
+        ):
+            if rank == 1:
+                color = GOLD_BOLD
+            elif rank <= 3:
+                color = RED_BOLD if rank % 2 == 0 else GREEN_BOLD
+            else:
+                color = RED if rank % 2 == 0 else GREEN
+
+            stars_str = f"⭐ ({stars})"
+            if stars < 10:
+                stars_str += " "
+
+            line = f"{color}{rank:>3}  | {stars_str:<7}| {score:>5} | {name}{RESET}"
+            lines.append(line)
+
+        embed = Embed(colour=Colors.dark_green.value)
+        embed.description = "```ansi\n" + "\n".join(lines) + "\n```"
+
+        return embed
+
+    @aoc.command(description="View aoc leaderboard.")
+    @option(
+        "year",
+        descrption="What year to fetch the leaderboard for.",
```

Comment:
Because of the TTL you can't guarantee fetch, more accurate would be
```suggestion
        descrption="What year to view the leaderboard for.",
```